### DISCUSSION TITLE
Multitenant: Errore di importazione questionari se il questionario è già presente in un altro sito

### DISCUSSION THREAD

**Discussion by verbavolant** [globaleaks/globaleaks-whistleblowing-software#3853] [General]
### What version of GlobaLeaks are you using?

4.13.18

### What operating system(s) are you seeing the problem on?

Linux

### Describe the issue

Ho generato un modello generico di questionario. Sistema multitenant.
Se provo ad importare il json appena esportato ottengo un errore "sqlalchemy.exc.IntegrityError Wraps a DB-API IntegrityError.", mi pare faccia l'importazione anche dell'id che è già presente in sql.

Soluzione temporanea: duplicare il modello da esportare, esportarlo in json, cancellare il modello appena duplicato, importarlo nel nuovo tenant, funziona.

**Comment by evilaliv3** [comment]
Grazie @verbavolant 

Se capisco mene tu stai cercando di importate su una istanza un questionario duplicato; potresti definirlo sulla primaria e ritrovartelo su tutte le istanze secondarie  riusarlo senza duplicazione.

Potremmo ovviamente automatizzare noi di generare ID univoci ogni volta che si fa esport, ma al momento non lo faccamo così che sistemi indipendenti possano condividere medesimi id e dunque in futuro statistiche.

Ben vengano i tuoi suggerimenti se vuoi proporre una modifica che credi migliorativa.

**Reply by verbavolant** [reply]
Ok perfetto. La duplicazione era per poter personalizzare alcune parti dei questionari (es nome della ditta nei form).
Modifica migliorativa potrebbe essere quella di inserire una variabile da mettere nei form del questionario per iniettare il nome dell'azienda e/o altre variabili?

Sarebbe forse meglio intanto implementare un errore più specifico magari?

Grazie!

**Reply by evilaliv3** [reply]
In merito agli errori, si ci dovrebbe lavorare.

Oggi GlobaLeaks per ragioni di resilienza implementa misure che prevengono situazioni problematiche (e.g. vincoli di integrità sul database) e per ragioni di risorse (è tradotto da volontari in 90 lingue) il numero di testi inclusi nell'applicazione è mantenuto baso (siamo a circa 700 stringhe).

Man mano che la community evolverà si spera cresca il numero di contribuitori e di risorse messe a disposizione per queste migliorie.
